src/lib/core/media-marshaller/media-marshaller.ts

```diff
@@ -326,6 +330,22 @@ export class MediaMarshaller {
     return (key === undefined || lastHope && lastHope.has(key)) ? lastHope : undefined;
   }
 
+  /**
+   * get the bpMap's value for the current breakpoint on the server
+   * @param bpMap
+   * @param key
+   */
+  private getServerActivatedValues(bpMap: BreakpointMap, key?: string): ValueMap | undefined {
+    // Unlike on the browser, only the currently activated breakpoint should be used on the server
```

Comment:
This is what I don't understand about your PR; I'm hoping you can clarify. The _intention_ is to generate all possible breakpoint styles on the server, because you don't _know_ what breakpoint is active on the client. The server has no active breakpoints! To that end, there's no concept of a browser, browser window, etc. We use the benefit of CSS (specifically the "Cascading") to give priority to the more exclusive breakpoints. If there's an issue there, the generation algorithm is in the `server` package. But I don't see how this fixes anything.

Reply:
Haha. That makes two of us 😂

This is just based on the patch @wsfriday-sfd created in issues 1065. I don't get how, but universal rendering's been working on my own project ever since I used my own version with @wsfriday-sfd's patch. I was hoping just submitting it as a PR would allow me to upgrade and keep up beyond Angular 6, but it seems it might not actually be solving the underlying problem.

If @wsfriday-sfd is around, perhaps they could clarify how/why this works in practice. Otherwise, I'll try to take a stab at actually understanding and solving where the bug is. Unfortunately I don't have tons of free time, but I'll try to look into it again in the next couple weeks.